src/mqt/problemsolver/partialcompiler/qaoa.py

```diff
@@ -0,0 +1,145 @@
+
+    def compile_qc(self, baseline: bool = False, opt_level: int = 3) -> QuantumCircuit:
+        """Compiles the circuit"""
+        if baseline:
+            return transpile(self.qc_baseline, backend=self.backend, optimization_level=opt_level, seed_transpiler=42)
+        return transpile(self.qc, backend=self.backend, optimization_level=opt_level, seed_transpiler=42)
+
+    def get_to_be_removed_gate_indices(self) -> list[int]:
+        """Returns the indices of the gates to be removed"""
+        indices_parameterized_gates = []
+        for i, gate in enumerate(self.qc_compiled._data):
+            if (
+                gate.operation.name == "rz"
+                and isinstance(gate.operation.params[0], Parameter)
+                and gate.operation.params[0].name.startswith("a_")
+            ):
+                indices_parameterized_gates.append(i)
+        assert len(indices_parameterized_gates) == len(self.remove_gates)
+        indices_to_be_removed_parameterized_gates = []
+        for i in range(len(indices_parameterized_gates)):
+            if self.qc_compiled._data[indices_parameterized_gates[i]].operation.params[0].name in self.remove_gates:
+                indices_to_be_removed_parameterized_gates.append(indices_parameterized_gates[i])
+
+        assert len(set(indices_to_be_removed_parameterized_gates)) == len({elem for elem in self.remove_gates if elem})
+        return indices_to_be_removed_parameterized_gates
+
+    def remove_unnecessary_gates(self, qc: QuantumCircuit, optimize_swaps: bool = True) -> QuantumCircuit:
+        """Removes the gates to be checked from the circuit at online time"""
+        to_be_deleted_indices = []
+
+        # Iterate over all gates to be removed
+        for i in self.to_be_removed_gates_indices:
+            # Remove the gate from the ParameterTable (was needed for QCEC to work)
+            del qc._parameter_table[
+                qc._data[i].operation.params[0]
+            ]  # remove the Parameter from the ParameterTable for the specific parameter. The table just contains one entry
+            if optimize_swaps and qc._data[i - 1].operation.name == "cx" and qc._data[i - 1] == qc._data[i + 1]:
+                to_be_deleted_indices.extend([i - 1, i, i + 1])
+            else:
+                to_be_deleted_indices.append(i)
+
+        for index in sorted(to_be_deleted_indices, reverse=True):
+            del qc._data[index]
+
+        return qc
+
+
+def get_backend(num_qubits: int) -> FakeBackend:
+    FakeManila().configuration()
+    montreal_config = FakeMontreal().configuration()
+    washington_config = FakeWashington().configuration()
+    manila_config = FakeManila().configuration()
+    FakeManila().configuration()
+    if num_qubits <= manila_config.n_qubits:
+        return FakeManila()
+    if num_qubits <= montreal_config.n_qubits:
+        return FakeMontreal()
+    if num_qubits <= washington_config.n_qubits:
+        return FakeWashington()
+    return None
```

Comment:
could be optimized. Untested suggestion:
```suggestion
    
    manila = FakeManila()
    if num_qubits <= manila.configuration().n_qubits:
        return manila

    montreal = FakeMontreal()
    if num_qubits <= montreal.configuration().n_qubits:
        return montreal
        
    washington = FakeWashington()
    if num_qubits <= washington.configuration().n_qubits:
        return washington
    
    return None
```